sdk/servicebus/azure-servicebus/azure/servicebus/management/_utils.py

```diff
@@ -326,14 +321,18 @@ def _validate_topic_subscription_and_rule_types(
         )
 
 def create_properties_from_dict_if_needed(properties, sb_resource_type):
-    # type: (DictPropertiesType, type) -> DictPropertiesReturnType
+    # type: (Union[PropertiesType, Mapping[str, Any]], Type[PropertiesType]) -> PropertiesType
     """
     This method is used to create a properties object given the
     resource properties type and its corresponding dict representation.
     :param properties: A properties object or its dict representation.
-    :type properties: DictPropertiesType
+    :type properties: Mapping or PropertiesType
     :param type sb_resource_type: The type of properties object.
-    :rtype: DictPropertiesReturnType
+    :rtype: PropertiesType
     """
-    return_properties = sb_resource_type(**properties) if isinstance(properties, dict) else properties
-    return return_properties
+    if isinstance(properties, sb_resource_type):
+        return properties
+    try:
+        return sb_resource_type(**cast(Mapping[str, Any], properties))
+    except TypeError:
+        raise TypeError("Update input must be an instance of {}, or a mapping".format(sb_resource_type.__name__))
```

Comment:
in the constructor of `*Properties`, a method called `validate_extraction_missing_args` will be called to check if there's any required keyword argument missing, if there is, then it will also raise TypeError, wouldn't that message be swallowed and replaced by this `TypeError("Update input must be an instance of {}, or a mapping".format(sb_resource_type.__name__))`?

the code 
```
def validate_extraction_missing_args(extraction_missing_args):
    if extraction_missing_args:
        raise TypeError(
            "__init__() missing {} required keyword arguments: {}".format(
                len(extraction_missing_args),
                " and ".join(["'" + e + "'" for e in extraction_missing_args]),
            )
        )
```

Reply:
Yeah this is a really good point - I will sleep on it...

Reply:
I think what we could do is:

1. have a separate instance check at the front:
```python
if not isinstance(properties, Mapping):
    raise TypeError("Update input must be an instance of {}, or a mapping".format(sb_resource_type.__name__))
return sb_resource_type(**cast(Mapping[str, Any], properties))
```
2. in the except block check the message of the exception:
```python
except TypeError as e:
    if "missing" in str(e):
        raise e
    raise TypeError("Update input must be an instance of {}, or a mapping".format(sb_resource_type.__name__))
```

Reply:
Unfortunately there's no such thing as a type check for a protocol.... as this kind of negates the reason to accept a protocol :)
I suspect option 2 would be the only way to surface the original error message.